runtime/sema/type.go

```diff
@@ -3734,23 +3741,33 @@ func (t *CompositeType) Tag() TypeTag {
 	return CompositeTypeTag
 }
 
-func (t *CompositeType) ExplicitInterfaceConformanceSet() *InterfaceSet {
-	t.initializeExplicitInterfaceConformanceSet()
-	return t.explicitInterfaceConformanceSet
+func (t *CompositeType) EffectiveInterfaceConformanceSet() *InterfaceSet {
+	t.initializeEffectiveInterfaceConformanceSet()
+	return t.effectiveInterfaceConformanceSet
 }
 
-func (t *CompositeType) initializeExplicitInterfaceConformanceSet() {
-	t.explicitInterfaceConformanceSetOnce.Do(func() {
-		// TODO: also include conformances' conformances recursively
-		//   once interface can have conformances
+func (t *CompositeType) initializeEffectiveInterfaceConformanceSet() {
+	t.effectiveInterfaceConformanceSetOnce.Do(func() {
+		t.effectiveInterfaceConformanceSet = NewInterfaceSet()
 
-		t.explicitInterfaceConformanceSet = NewInterfaceSet()
-		for _, conformance := range t.ExplicitInterfaceConformances {
-			t.explicitInterfaceConformanceSet.Add(conformance)
+		for _, conformance := range t.EffectiveInterfaceConformances() {
+			t.effectiveInterfaceConformanceSet.Add(conformance.InterfaceType)
 		}
 	})
 }
 
+func (t *CompositeType) EffectiveInterfaceConformances() []Conformance {
```

Comment:
Just curious: Is it necessary we keep both a set and a slice of the effective interface conformances?

Reply:
This is pretty much the same as what's there currently:
```go
explicitInterfaceConformanceSet     *InterfaceSet
ExplicitInterfaceConformances       []*InterfaceType
```

We could construct the slice the first time, but that would undo some other optimizations, like unnecessarily constructing a set, when it is not actually needed, etc. Not sure which is more expensive tbh, i.e: storing the interim result (the slice) vs always constructing the set.